test/e2e/e2e_boyar_test.go

```diff
@@ -131,3 +131,31 @@ func TestE2EAddNewVirtualChainWithSwarmAndBoyar(t *testing.T) {
 		helpers.WaitForBlock(t, helpers.GetMetricsForPort(8175), 0, WAIT_FOR_BLOCK_TIMEOUT)
 	})
 }
+
+// Tests boyar.Flow as close as it gets to production starting up
+func TestE2EWithFullFlowAndDisabledSimilarVchainId(t *testing.T) {
+	withCleanContext(t, func(t *testing.T) {
+		for i := 1; i <= 3; i++ {
+			vchains := getBoyarVchains(i, 1000, 92, 100)
+			vchains[len(vchains) - 1].Disabled = true // Check for namespace clashes: 100 will be removed but 1000 should be intact
+
+			boyarConfig := getBoyarConfig(vchains)
+			cfg, err := config.NewStringConfigurationSource(string(boyarConfig), "")
+			cfg.SetKeyConfigPath(fmt.Sprintf("%s/node%d/keys.json", getConfigPath(), i))
+			require.NoError(t, err)
+
+			logger := helpers.DefaultTestLogger()
+			cache := config.NewCache()
+			err = boyar.Flow(context.Background(), cfg, cache, logger)
+			require.NoError(t, err)
+		}
+
+		helpers.WaitForBlock(t, helpers.GetMetricsForPort(9081), 3, WAIT_FOR_BLOCK_TIMEOUT)
+		helpers.WaitForBlock(t, helpers.GetMetricsForPort(8175), 0, WAIT_FOR_BLOCK_TIMEOUT)
+
+		helpers.Eventually(WAIT_FOR_BLOCK_TIMEOUT, func() bool {
```

Comment:
after employing some detective work, I understand that this checks that chain 100 is offline? if so, can you please make this assertion clearer?